src/App/Fossa/IAT/Fingerprint.hs

```diff
@@ -0,0 +1,39 @@
+module App.Fossa.IAT.Fingerprint (
+  fingerprintContentsRaw,
+) where
+
+import App.Fossa.IAT.Types
+import Control.Carrier.Diagnostics
+import Control.Effect.Lift
+import Crypto.Hash
+import Data.ByteString.Lazy qualified as BL
+import Data.String.Conversion
+import Data.Text
+import Discovery.Walk
+import Effect.ReadFS
+import Path
+
+sha256 :: BL.ByteString -> Digest SHA256
+sha256 = hashlazy
+
+hashFile :: (Has (Lift IO) sig m, Has Diagnostics sig m) => FilePath -> m Text
+hashFile file = do
+  content <- context "Read file" $ sendIO $ BL.readFile file
```

Comment:
I don't think we should be doing raw, lazy IO in production code.   I think that the `ReadFS` effect needs to be able to produce some sort of streaming type.  @cnr what do you think here?

Reply:
Yeah frustratingly here, `BL.readFile` isn't a safe operation

For streaming elsewhere we use [conduit](https://github.com/snoyberg/conduit#readme), and it looks like there's a package `conduit-cryptonite` that [offers a hashFile function](https://hackage.haskell.org/package/cryptonite-conduit-0.2.2/docs/Crypto-Hash-Conduit.html#v:hashFile) -- which can be used as `sendIO $ hashFile ...`

The body of that function is [pretty small](https://hackage.haskell.org/package/cryptonite-conduit-0.2.2/docs/src/Crypto-Hash-Conduit.html) if we'd like to just reimplement it as well

The other piece to note here is that errors aren't going to be great, because failures will just throw `IOException` and entirely skip our `Diagnostics` effect. We could resolve that by using `catch` from `Control.Effect.Exception` and catching `IOException` specifically to rethrow with `fatal`:

```hs
hashFile = ...
  `catch` (\(e :: IOException) -> fatalText ("blah blah failed to hash file " <> toText (show e)))
```

Reply:
@cnr 

> The body of that function is pretty small if we'd like to just reimplement it as well

What is the typical practice within Spectrometer? I'm happy to go either route.
Also the library you're referring to is [this one](https://hackage.haskell.org/package/cryptonite-conduit), right?

Reply:
I'd re-implement it rather than pulling in the dep; the function is really small. And yes, that's the one